lib/xml/reader.dart

```diff
@@ -141,3 +144,147 @@ class XmlReader {
     return context.success(null, context.position + 1);
   }
 }
+
+/// A push based XmlReader interface, intended to be similar to .NET's XmlReader.
```

Comment:
I am not familiar with .NET, could you expand on why this is different to the above? Also, this looks to me rather like pull-based, and it is not clear to me why it is called text-reader?

Reply:
For a SAX/pull based reader, you call `parse` and it calls you back.  For this reader, you feed it the text and then call `read` when you're ready to move forward.  `read` will update the state of the parser.

I'm not married to the name - in .NET, `XmlReader` is this style, and a concrete implementation is called `XmlTextReader` - there's probably a better way to name it.